spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -266,4 +266,9 @@ public boolean isRunning() {
 	public int getPhase() {
 		return 0;
 	}
+
+	@Override
+	public void destroy() throws Exception {
```

Comment:
This will call the doTaskEnd twice at the end of a task that ends normally.  Once for the onApplicationEvent then again for the destroy.  
Do we want to put a flag here that will prevent it from calling the destroy if we have already recorded our results?